OpenRA.Game/MiniYaml.cs

```diff
@@ -119,6 +119,24 @@ public sealed class MiniYaml
 		public readonly string Value;
 		public readonly ImmutableArray<MiniYamlNode> Nodes;
 
+		public bool Matches(MiniYaml other)
+		{
+			if (other == null || Nodes.Length != other.Nodes.Length
+				|| !Value.Equals(other.Value, StringComparison.InvariantCulture))
+				return false;
+
+			for (var i = 0; i < Nodes.Length; i++)
+			{
+				var n = Nodes[i];
+				var on = other.Nodes[i];
+				if (!n.Key.Equals(on.Key, StringComparison.InvariantCulture)
+					|| !n.Value.Matches(on.Value))
+					return false;
+			}
+
+			return true;
+		}
```

Comment:
it looks like `SequenceEqual` is also using a for loop so I think this implementation is cleaner than adding an `IEqualityComparer`

https://github.com/steveharter/dotnet_corefx/blob/master/src/System.Linq/src/System/Linq/SequenceEqual.cs

Reply:
I see no null checks on key in the constructors. Since this is a lower level generic object. Perhaps it would be worth doing a null check on key.

Reply:
Same for value. Value can never be null?

Reply:
Well it didn't crash on any of the maps I tried to run